lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -339,103 +357,88 @@ void ThreadManager::Impl::addWorker(size_t value) {
     idMap_.insert(std::pair<const Thread::id_t, shared_ptr<Thread> >((*ix)->getId(), *ix));
```

Comment:
I feel the changes in here are substantial and quite useful, in that I have added tests and re-enabled a number of tests, and ctest on windows runs in about a minute now.  As for getting rid of idMap_ and canSleep, I see no reason to keep them.  As for 0.10.0, all of these changes are pretty substantial and I don't know if they make it in or not.  I think they are substantial improvements in correctness and stability, and should be considered at least.